appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -2,448 +2,421 @@ package nu.yona.server
+	def 'Richard processes Bob\'s buddy acceptance'()
+	{
 		given:
+			def richard = addRichard()
+			def mobileNumberBob = "+$timestamp"
+			def inviteURL = sendBuddyRequestForBob(richard, mobileNumberBob).responseData.userCreatedInviteURL
+			def bob = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBoddyRequest, inviteURL, true, null)
+			def newNickname = "Bobby"
+			def newPassword = "B o b"
+			def updatedBobJson = bob.convertToJSON()
+			updatedBobJson.nickname = newNickname
+			updatedBobJson.devices = [ "iPhone 6" ]
+			updatedBobJson.goals = [ "gambling" ]
+			def updatedBob = appService.updateUser(appService.&assertUserUpdateResponseDetails, new User(updatedBobJson, newPassword), inviteURL);
+			appService.confirmMobileNumber(appService.&assertResponseStatusSuccess, updatedBob)
+			def acceptURL = appService.fetchBuddyConnectRequestMessage(updatedBob).acceptURL
+			appService.postMessageActionWithPassword(acceptURL, ["message" : "Yes, great idea!"], newPassword)
 
 		when:
-			def response = appService.updateUser(bobDunnInviteURL, """{
-				"firstName":"Bob ${timestamp}",
-				"lastName":"Dun ${timestamp}",
-				"nickname":"BD ${timestamp}",
-				"mobileNumber":"+${timestamp}13",
-				"devices":[
-					"iPhone 6"
-				],
-				"goals":[
-					"gambling"
-				]
-			}""", bobDunnPassword)
-			if(response.status == 200)
-			{
-				bobDunnURL = appService.stripQueryString(response.responseData._links.self.href)
-				bobDunnMobileNumberConfirmationCode = response.responseData.confirmationCode;
-			}
+			def processURL = appService.fetchBuddyConnectResponseMessage(richard).processURL
+			def response = appService.postMessageActionWithPassword(processURL, [ : ], richard.password)
 
 		then:
 			response.status == 200
-			bobDunnMobileNumberConfirmationCode
-			response.responseData.firstName == "Bob ${timestamp}"
-			response.responseData.lastName == "Dun ${timestamp}"
-			response.responseData.mobileNumber == "+${timestamp}13"
-			response.responseData.nickname == "BD ${timestamp}"
-			response.responseData.devices.size() == 1
-			response.responseData.devices[0] == "iPhone 6"
-			//TODO: updating of goals is not yet supported
-			response.responseData.goals.size() == 0
+			response.responseData.properties.status == "done"
+
+			def richardWithBuddy = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBoddyRequest, richard.url, true, richard.password)
+			richardWithBuddy.buddies != null
+			richardWithBuddy.buddies.size() == 1
+			richardWithBuddy.buddies[0].user.firstName == "Bob"
+			richardWithBuddy.buddies[0].sendingStatus == "ACCEPTED"
+			richardWithBuddy.buddies[0].receivingStatus == "ACCEPTED"
+
+		cleanup:
+			appService.deleteUser(richard)
+			appService.deleteUser(updatedBob)
 	}
 
-	def 'Bob cannot read direct messages before confirming mobile number'(){
+	def 'Goal conflict of Richard is reported to Richard and Bob'()
```

Comment:
Same here. The alternative would be to do these actions in the ``then`` block. Till now, I didn't do that but only put verficiation/read actions in the ``then`` blocks.